core/src/com/unciv/logic/battle/BattleDamage.kt

```diff
@@ -75,12 +75,16 @@ object BattleDamage {
                 }
             }
 
-            if (civInfo.hasUnique("+15% combat strength for melee units which have another military unit in an adjacent tile")
-                && combatant.isMelee()
-                && combatant.getTile().neighbors.flatMap { it.getUnits() }
+            if (combatant.getTile().neighbors.flatMap { it.getUnits() }
                     .any { it.civInfo == civInfo && !it.type.isCivilian() && !it.type.isAirUnit() }
-            )
-                modifiers["Discipline"] = 15
+            ) {
+                var adjacentUnitBonus = 0;
+                for (unique in civInfo.getMatchingUniques("+[] combat strength for [] units which have another military unit in an adjacent tile")) {
+                    if ((combatant.isMelee() && unique.params[1] == "melee") || (combatant.isRanged() && unique.params[1] == "ranged"))
```

Comment:
`combatant.unit.matchesFilter(unique.params[1])`? a) the [] notation suggests the standard unitFilter is supported, so maybe we should support it; b) shorter to write    ...?

Reply:
Fixed in #4118